transitions/registration.js

```diff
@@ -217,21 +216,29 @@ module.exports = {
                 if (self.isBoolExprFalse(obj, event.bool_expr)) {
                     return;
                 }
-                var options = { db: db, audit: audit, doc: doc };
-                if (event.params) {
-                    // params setting get sent as array
+                var options = { db: db, audit: audit, doc: doc, registrationConfig: registrationConfig };
+
+                if (typeof event.params === 'string') {
+                    // params setting can get sent as string to convert into an array
                     options.params = event.params.split(',');
+                } else if (typeof event.params === 'object') {
```

Comment:
I see you have unit tests for this, but have you tried using an object on an actual instance? In the kanso.json app_settings schema event params are `"type": "string"` which, looking through the code, should make this branch unreachable. It's definitely a good idea to allow JSON as event params but we might need to JSON.parse it somewhere first.